store/postgres/src/store.rs

```diff
@@ -765,6 +668,128 @@ impl StoreTrait for Store {
+                            .expect("invalid subgraph name found in database");
+                        let subgraph_id = SubgraphId::new(subgraph_id)
+                            .expect("invalid subgraph ID found in database");
+                        (name, subgraph_id)
+                    }).collect()
+            })
+    }
+
+    fn write(
+        &self,
+        name: SubgraphDeploymentName,
+        subgraph_id: SubgraphId,
+        node_id: NodeId,
+    ) -> Result<(), Error> {
+        use db_schema::subgraph_deployments;
+
+        insert_into(subgraph_deployments::table)
+            .values((
+                subgraph_deployments::deployment_name.eq(name.to_string()),
+                subgraph_deployments::subgraph_id.eq(subgraph_id.to_string()),
+                subgraph_deployments::node_id.eq(node_id.to_string()),
+            )).on_conflict(subgraph_deployments::deployment_name)
+            .do_update()
+            .set((
+                subgraph_deployments::subgraph_id.eq(subgraph_id.to_string()),
+                subgraph_deployments::node_id.eq(node_id.to_string()),
+            )).execute(&*self.conn.lock().unwrap())
+            .map_err(Error::from)
+            .map(|_| ())
+    }
+
+    fn read(&self, name: SubgraphDeploymentName) -> Result<Option<(SubgraphId, NodeId)>, Error> {
+        use db_schema::subgraph_deployments;
+
+        subgraph_deployments::table
+            .select((
+                subgraph_deployments::subgraph_id,
+                subgraph_deployments::node_id,
+            )).filter(subgraph_deployments::deployment_name.eq(name.to_string()))
+            .first::<(String, String)>(&*self.conn.lock().unwrap())
+            .optional()
+            .map_err(Error::from)
+            .map(|row_opt| {
+                row_opt.map(|(subgraph_id, node_id)| {
+                    let subgraph_id = SubgraphId::new(subgraph_id)
+                        .expect("invalid subgraph ID found in database");
+                    let node_id = NodeId::new(node_id).expect("invalid node ID found in database");
+                    (subgraph_id, node_id)
+                })
+            })
+    }
+
+    fn remove(&self, name: SubgraphDeploymentName) -> Result<bool, Error> {
+        use db_schema::subgraph_deployments;
+
+        delete(subgraph_deployments::table)
+            .filter(subgraph_deployments::deployment_name.eq(name.to_string()))
+            .execute(&*self.conn.lock().unwrap())
+            .map(|row_count| match row_count {
+                0 => false,
+                1 => true,
+                _ => unreachable!(),
+            }).map_err(Error::from)
+    }
+
+    fn deployment_events(
+        &self,
+        node_id: NodeId,
+    ) -> Box<Stream<Item = DeploymentEvent, Error = Error> + Send> {
+        // Create postgres listener
+        let channel_name =
+            SafeChannelName::i_promise_this_is_safe(format!("subgraph_deployment_{}", node_id));
```

Comment:
This opens up `node_id` as an attack vector, right? While this naming scheme is clever, if it's a potential security issue then we should come up with something else. What about generating an UUID for the channel and maintaining a table that maps node_ids to channel ids?

Reply:
I just added some comments to explain why I think it's safe. Any node IDs that would cause a problem won't pass validation. I also added a comment to `NodeId::new` where the validation happens mentioning why the validation is important, so nobody removes it without thinking about the consequences. Do you think that's good enough?

Reply:
Here's the `NodeId::new` comment:

https://github.com/graphprotocol/graph-node/pull/606/files#diff-af969dcb4d984da0c9a0effb86b21305R75

Reply:
Ah nice, thanks for the explanation and the added comment.